src/ott/solvers/nn/neuraldual.py

```diff
@@ -12,73 +12,114 @@
 """A Jax implementation of the ICNN based Kantorovich dual."""
 
 import warnings
-from typing import Dict, Iterable, List, Literal, Optional, Tuple, Union
+from typing import Callable, Dict, Iterable, List, Literal, Optional, Tuple, Union
 
-import flax.linen as nn
 import jax
 import jax.numpy as jnp
 import optax
 from flax import core
+from flax.core import frozen_dict
 
 from ott.geometry import costs
 from ott.problems.linear import potentials
-from ott.solvers.nn import icnn
+from ott.solvers.nn import conjugate_solver, icnn, potential_base
 
-__all__ = ["NeuralDualSolver"]
+__all__ = ["W2NeuralDual"]
 
 Train_t = Dict[Literal["train_logs", "valid_logs"], Dict[str, List[float]]]
-
-
-class NeuralDualSolver:
-  r"""Solver of the ICNN-based Kantorovich dual.
-
-  Learn the optimal transport between two distributions, denoted source
-  and target, respectively. This is achieved by parameterizing the two
-  Kantorovich potentials, `g` and `f`, by two input convex neural networks.
-  :math:`\nabla g` hereby transports source to target cells, and
-  :math:`\nabla f` target to source cells.
-
-  Original algorithm is described in :cite:`makkuva:20`.
+Callback_t = Callable[[int, potentials.DualPotentials], None]
+Conj_t = Optional[conjugate_solver.FenchelConjugateSolver]
```

Comment:
No need for this alias, used only once.

Reply:
I originally did this because unfortunately inlining it seems to create a bug between `autoflake` and `flake8`: `autoflake` formats it to:

![image](https://user-images.githubusercontent.com/707462/217891496-4946fd2e-b0a7-4212-a8fa-5eaaca38601b.png)

but then flake gives the error:

![image](https://user-images.githubusercontent.com/707462/217891535-50d40e07-94da-406e-bf3d-5c63cf180b1f.png)

And any manual edits I do to the line get automatically reverted with `autoflake`. Do you have any other ideas to fix it?

Reply:
Ok, let's keep it as it is then!